arch/inst/C/c.ld.yaml

```diff
@@ -32,16 +33,20 @@ access:
   vu: always
 operation(): |
   if (implemented?(ExtensionName::C) && (CSR[misa].C == 1'b0)) {
-    raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+      raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
   }
 
-  XReg virtual_address = X[creg2reg(rs1)] + imm;
-
-  X[creg2reg(rd)] = sext(read_memory<64>(virtual_address, $encoding), 64);
-
-# SPDX-SnippetBegin
-# SPDX-FileCopyrightText: 2017-2025 Contributors to the RISCV Sail Model <https://github.com/riscv/sail-riscv/blob/master/LICENCE>
-# SPDX-License-Identifier: BSD-2-Clause
+  if (XLEN == 32) {
+      if (implemented?(ExtensionName::Zclsd)) {
+          X[rd] = sext(read_memory<64>(X[rs1] + imm, $encoding), 64);
+          X[rd + 1] = sext(read_memory<64>(X[rs1] + imm + 8, $encoding), 64);
```

Comment:
use `creg2reg` for any references to registers with compressed instructions, so `creg2reg(rd)`, for example.